### PR TITLE
nixos.system.boot.kernel: stronger constraints for kernelParam type

### PR DESCRIPTION
**Pull request by cab404**
Checks whether all spaces are inside double quotes, thus ensuring that one
string parses as no more than one kernel param.

###### Motivation for this change

Users often add spaces in expressions like "a = 12", and having them parsed as 3 separate params is not user friendly.

###### Things done

kernelParams type changed to a string match.

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [x] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: nixos/modules/system/boot/kernel.nix [hunk 1]
```diff
@@ -83,7 +83,10 @@ in
     };
 
     boot.kernelParams = mkOption {
-      type = types.listOf types.str;
+      type = types.listOf (types.strMatching ''([^" ]*|"[^"]*")*'' // {
```
Comment (pennae) [inline nixos/modules/system/boot/kernel.nix:86]: this should check for all code points `isspace()` detects as spaces. maybe for all spaces in general, better to be a little overzealous in having people add quotes than create bad command lines

Comment (cab404) [reply nixos/modules/system/boot/kernel.nix:86]: sounds reasonable, will change to \s when back home

**Comment by cab404** [comment]
It actually contains some breaking changes (e.g some of the old configs won't build), so I will need to update release notes.

**Review by tomberek — APPROVED** [review]

File: nixos/doc/manual/release-notes/rl-2111.section.md [hunk 2]
```diff
@@ -171,7 +171,6 @@ pt-services.clipcat.enable).
 
 - `programs.neovim.runtime` switched to a `linkFarm` internally, making it impossible to use wildcards in the `source` argument.
 
-<<<<<<< HEAD
```
Comment (Artturin) [inline nixos/doc/manual/release-notes/rl-2111.section.md:174]: You forgot this

**Review by nrdxp — APPROVED** [review]
Seems to work as intended when evaluating a system from this branch of nixpkgs. Just rebase and you've got my approval to merge :+1:

**Review by pennae — APPROVED** [review]

**Comment by nrdxp** [comment]
There is no demerge button unfortunately, but I'll gladly merge a doc PR :sweat_smile: 
Sorry, I guess I got a little antsy :sweat:

**Comment by cab404** [comment]
> There is no demerge button unfortunately, but I'll gladly merge a doc PR sweat_smile Sorry, I guess I got a little antsy sweat

#141818 :heart:

**Pull request merged**